xblock/fields.py

```diff
@@ -779,6 +779,18 @@ def from_json(self, value):
 
     enforce_type = from_json
 
+    def to_string(self, value):
+        """
+        In python3, json.dumps() cannot sort keys of different types, 
+        so preconvert None to 'null'.
+        """
+        self.enforce_type(value)
+        if isinstance(value, dict) and None in value:
+            value = value.copy()
+            value['null'] = value[None]
+            del value[None]
+        return super(Dict, self).to_string(value)
```

Comment:
@cpennington We have a test that checks that things work with null keys.  Do we also allow numeric keys?  Because those would also fail for the same reason (different types not comparable in python3), and we might need to do the same kind of processing for them here.  

However, with None, we can perform the fix in constant time, whereas for numbers, we would have to process each key in turn, so I didn't want to do the work unless there's a use for it.